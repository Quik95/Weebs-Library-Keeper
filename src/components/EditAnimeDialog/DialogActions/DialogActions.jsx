import React, { memo } from 'react';
import PropTypes from 'prop-types';

// MUI
import MuiDialogActions from '@material-ui/core/DialogActions';
import Button from '@material-ui/core/Button';
import DeleteIcon from '@material-ui/icons/Delete';
import CloseIcon from '@material-ui/icons/Close';
import SaveIcon from '@material-ui/icons/Save';

// styles
import useStyles from './DialogActions.styles';
import clsx from 'clsx'

const DialogActions = memo(function DialogActions({ handleClose }) {
  const classes = useStyles();

  return (
    <MuiDialogActions className={classes.dialogActions}>
      <Button
        variant="contained"
        startIcon={<DeleteIcon />}
        className={clsx(classes.deleteButton, classes.button)}
      >
        Delete
      </Button>
      <Button variant="contained" color="primary" className={classes.button} startIcon={<CloseIcon />}>
        Discard
      </Button>
      <Button
        variant="contained"
        startIcon={<SaveIcon />}
        className={clsx(classes.saveButton, classes.button)}
      >
        Save
      </Button>
    </MuiDialogActions>
  );
});

DialogActions.propTypes = {
  handleClose: PropTypes.func.isRequired,
};

export default DialogActions;
