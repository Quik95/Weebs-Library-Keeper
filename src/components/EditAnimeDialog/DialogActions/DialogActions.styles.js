import { makeStyles } from '@material-ui/core/styles';

import { green, red } from '@material-ui/core/colors';

export default makeStyles(theme => ({
  dialogActions: {
    display: 'grid',
    gridTemplateColumns: '1fr 120px 120px',
  },
  saveButton: {
    backgroundColor: green[500],
    color: 'white',
    '&:hover': {
      backgroundColor: green[700],
    },
  },
  deleteButton: {
    backgroundColor: red[500],
    justifySelf: 'start',
    color: 'white',
    '&:hover': {
      backgroundColor: red[700],
    },
  },
}));
