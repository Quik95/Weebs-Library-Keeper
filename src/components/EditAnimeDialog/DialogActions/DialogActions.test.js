import DialogActions from './DialogActions';

describe('<DialogActions/>', () => {
  const handleCloseMock = jest.fn(() => {});

  const wrapper = shallow(<DialogActions handleClose={handleCloseMock} />);

  it('Renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
