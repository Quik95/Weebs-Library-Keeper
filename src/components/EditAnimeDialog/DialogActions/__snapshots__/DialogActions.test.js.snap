// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DialogActions/> Renders correctly 1`] = `
<WithStyles(ForwardRef(DialogActions))
  className="makeStyles-dialogActions-1"
>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-deleteButton-4 makeStyles-button-2"
    startIcon={<Memo(DeleteIcon) />}
    variant="contained"
  >
    Delete
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-button-2"
    color="primary"
    startIcon={<Memo(CloseIcon) />}
    variant="contained"
  >
    Discard
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button))
    className="makeStyles-saveButton-3 makeStyles-button-2"
    startIcon={<Memo(SaveIcon) />}
    variant="contained"
  >
    Save
  </WithStyles(ForwardRef(Button))>
</WithStyles(ForwardRef(DialogActions))>
`;
